### PR TITLE
Swap the volunteer role for the onboarding role

### PR DESCRIPTION
**Pull request by Adimote**
Also dont show the welcome message for blueshirts, as we now have
to wait until they finish onboarding before we can announce them.

This is to handle the new onboarding process we have for blueshirts

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -113,27 +112,26 @@ async def on_message(message: discord.Message) -> None:
...
+
+        if chosen_team != SPECIAL_TEAM:
+            announce_channel: discord.TextChannel = discord.utils.get(
+                message.guild.channels,
+                name=ANNOUNCE_CHANNEL_NAME,
+            )
+            await announce_channel.send(
+                f"Welcome {message.author.mention} from team {chosen_team}",
+            )
+            logger.info(f"Sent welcome announcement for '{message.author.name}'")
```
Comment (PeterJCLaw) [inline main.py:134]: Adding a blank line here would help visually clarify the end of the block:
```suggestion
            logger.info(f"Sent welcome announcement for '{message.author.name}'")

```

**Review by PeterJCLaw — COMMENTED** [review]
These changes seem fine, though I'm not really sure how to review them without more context.
I'd also suggest rebasing this PR so that the history is a bit more tractable.

**Comment by Adimote** [comment]
Here's a bit more context.

I added a new step between someone entering the password and getting the 'Blue Shirt' role, they first get the 'future-volunteer' role, which gives them access to the channel which has the onboarding documentation (i.e. turn off DMs, etc..)

That means this bot needs to give them the 'future-volunteer' role, they then can upgrade to Blue Shirt by confirming they've read the rules (Which is a different process, not used by this bot)

**Comment by PeterJCLaw** [comment]
(rebasing against `origin/main` would also let you pick up the fix for the requirements CI failure)

**Comment by PeterJCLaw** [comment]
I whur? Why are we re-applying the single-version commit here?

**Comment by Adimote** [comment]
I don't know what I did.

**Review by PeterJCLaw — APPROVED** [review]

**Pull request merged**